packages/components/src/hooks/usePasswordValidation.tsx

```diff
@@ -0,0 +1,115 @@
+import { Box, Flex, Text } from "@chakra-ui/react";
+import { useEffect, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import zxcvbn from "zxcvbn";
+
+const DEFAULT_SCORE = 0;
+const DEFAULT_COLOR_SCHEME = "gray.100";
+
+type PasswordStrengthBarProps = {
+  score: number;
+  colorScheme: string;
+  inputName: string;
+};
+
+const PasswordStrengthBar = ({ score, colorScheme, inputName }: PasswordStrengthBarProps) => {
+  const form = useFormContext();
+
+  const colors = [colorScheme, "red.500", "yellow.500", "green.500"];
+  const passwordError = form.formState.errors[inputName];
+
+  const getColor = (index: number) => {
```

Comment:
nit: maybe `getSectionColor` & `sectionIndex`?

Reply:
@OKendigelyan please consider renaming / adding a comment - it's a bit hard to understand what the function from a first glance